.github/workflows/empty-string-warning.yml

```diff
@@ -0,0 +1,77 @@
+name: Warn for Empty Strings
+
+on:
+  pull_request:
+    paths:
+      - "**/*.ts"
+      - "**/*.tsx"
+
+permissions:
+  issues: write
+  pull-requests: write
+
+jobs:
+  check-empty-strings:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Find empty strings in TypeScript files
+        id: find_empty_strings
+        run: |
+          # Find empty strings and mark them explicitly
+          output=$(grep -Rn --include=\*.ts "''\|\"\"" --exclude-dir={node_modules,dist,out} . || true)
+
+          if [ -z "$output" ]; then
+            echo "::set-output name=results::No empty strings found."
+            exit 0
+          else
+            output=$(echo "$output" | sed 's/""/"[EMPTY STRING]"/g' | sed "s/''/'[EMPTY STRING]'/g")
+            echo "::set-output name=results::${output//$'\n'/%0A}"
+          fi
+
+      - name: findings
+        if: steps.find_empty_strings.outputs.results != 'No empty strings found.'
+        run: |
+          if [ "${{ steps.find_empty_strings.outputs.results }}" == "No empty strings found." ]; then
+            echo "No empty strings found. No action required."
+          else
+            echo "::warning::Empty strings found in the following files:"
+            echo "${{ steps.find_empty_strings.outputs.results }}"
+          fi
+
+      - name: Find Comment
+        uses: peter-evans/find-comment@v3
+        id: find_comment
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body-includes: Empty String Usage Detected
+          comment-author: github-actions[bot]
+
+      - name: Comment on PR
+        if: steps.find_empty_strings.outputs.results != 'No empty strings found.'
+        uses: peter-evans/create-or-update-comment@v4
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-id: ${{ steps.find_comment.outputs.comment-id }}
+          edit-mode: replace
+          body: |
+            ## Empty String Usage Detected
+
+            The following occurrences of empty strings were detected. Please review them to ensure their usage is necessary:
+
+            ```plaintext
+            ${{ steps.find_empty_strings.outputs.results }}
+            ```
+
+      - name: Remove comment
+        if: steps.find_empty_strings.outputs.results == 'No empty strings found.'
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const { owner, repo } = context.repo;
+            const { number } = context.issue;
+            const comment_id = ${{ steps.find_comment.outputs.comment-id }}
+            github.rest.issues.deleteComment({ owner, repo, comment_id });
```

Comment:
These should only be annotated on the file view. No comments cluttering the conversation.

Reply:
comment on every individual file where the empty exists?

Reply:
We should also enable support for some type of ignore directive in a later task. I foresee that it will be annoying to have these annotations forever in the future even if we think the specific instance of the empty string is okay.

Reply:
So for every empty string in file, in the exact place it occurs in each file there should be one comment "Review this empty string" on file view and not in the conversational pr comments

Not one comment containing them all placed in the conversation and removed

Reply:
> So for every empty string in file, in the exact place it occurs in each file there should be one comment "Review this empty string" on file view and not in the conversational pr comments

Correct